### ISSUE TITLE
Ubuntu installation fails - 2 of the params to deploydb.sh are empty/blank

### ISSUE THREAD

**Issue by bmullan** [zstackorg/zstack#24]
Trying to install on Ubuntu 14.04 and after executing:

$ sudo bash install-zstack.sh -a

log file indicates 2 of the parameters being passed to deploydb.sh are blank ('') which from looking at the script causes that script to return a failure if ANY parameter is blank...

Deploy ZStack Database: ERROR: failed to execute shell command: sh /usr/local/zstack/apache-tomcat/webapps/zstack/WEB-INF/classes/deploydb.sh root '' 192.168.2.102 3306 ''
return code: 1
stdout: /usr/local/zstack/apache-tomcat/webapps/zstack/WEB-INF/classes/deploydb.sh root  192.168.2.102 3306

any suggestions on what to try next?

**Comment by youyk** [comment]
Do you install mysql before zstack installation? If yes, you need to pass
mysql root password to zstack-install.sh by -P, e.g.
sudo bash zstack-install.sh -a -P MY_MYSQL_ROOT_PASSWORD

On Wed, May 6, 2015 at 11:07 PM, bmullan <notifications@github.com> wrote:

> Trying to install on Ubuntu 14.04 and after executing:
>
> $ sudo bash install-zstack.sh -a
>
> log file indicates 2 of the parameters being passed to deploydb.sh are
> blank ('') which from looking at the script causes that script to return a
> failure if ANY parameter is blank...
>
> Deploy ZStack Database: ERROR: failed to execute shell command: sh
> /usr/local/zstack/apache-tomcat/webapps/zstack/WEB-INF/classes/deploydb.sh
> root '' 192.168.2.102 3306 ''
> return code: 1
> stdout:
> /usr/local/zstack/apache-tomcat/webapps/zstack/WEB-INF/classes/deploydb.sh
> root 192.168.2.102 3306
>
> any suggestions on what to try next?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zstackorg/zstack/issues/24>.
>



-- 
Yongkang You

**Comment by bmullan** [comment]
I was trying to do the quick install... but I decided to give up on that and do the manual install and that looks like it worked successfully just now.

Thanks
Brian

**Comment by youyk** [comment]
Great！

On Thu, May 7, 2015 at 1:09 AM, bmullan <notifications@github.com> wrote:

> I was trying to do the quick install... but I decided to give up on that
> and do the manual install and that looks like it worked successfully just
> now.
>
> Thanks
> Brian
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zstackorg/zstack/issues/24#issuecomment-99540100>.
>



-- 
Yongkang You

**Comment by bmullan** [comment]
I did figure out one thing that wasn't documented well or at least if it
was I didn't notice it.

Using the "quick install" was failing for me because I was installing it on
a new install machine that I had not installed openssh-server onto yet.

That apparently is a pre-req.

Also... a pre-req after installing openssh-server I had to change
"PermitRootLogin" from "no" to "yes"

I will open a separate "issue" about that required change though because I
think that requirement is a dangerous one for security for an IaaS platform.

Thanks

Brian
